### PR TITLE
Add Python 3.11 to support matrix

### PR DESCRIPTION
**Pull request by davidkyle**
Python 3.11 support was blocked by the version of PyTorch used. Now that PyTorch has been upgraded 3.12 can be added to the support matrix

PyTorch 2.2.0 supports Python 3.12 so we will have to wait until the next PyTorch upgrade to enable Python 3.12

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -27,9 +27,9 @@ steps:
     matrix:
       setup:
         python:
+          - '3.11'
           - '3.10'
           - '3.9'
-          - '3.8'
```
Comment (pquentin) [inline .buildkite/pipeline.yml:32]: Are you dropping Python 3.8 support on purpose? If yes we need to update `python_requires` in setup.py.

**Review by pquentin — COMMENTED** [review]
We need to update `CLASSIFIERS` in setup.py. And I would suggest running `pyupgrade --py38-plus **/*.py` or `pyupgrade --py39-plus **/*.py` depending on the actually mininum supported version.

The scientific ecosystem has mostly dropped Python 3.8, so I would be fine dropping it here too: https://scientific-python.org/specs/spec-0000/

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (davidkyle) [reply .buildkite/pipeline.yml:32]: I thought Python 3.8 could be removed from CI but as it is still supported that would be a silly thing to do. I've reverted this change

**Comment by davidkyle** [comment]
`CLASSIFIERS` is updated and the result of running `pyupgrade --py38-plus **/*.py` is in commit 04c38289d8cfe0b7f0c0fac34564424a64319f46

**Review by pquentin — APPROVED** [review]
Thanks! LGTM.

**Comment by davidkyle** [comment]
Thanks @pquentin

**Pull request merged**